SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialLoadBalancingHandler.cpp

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "Utils/SpatialLoadBalancingHandler.h"
+
+#include "EngineClasses/SpatialActorChannel.h"
+#include "EngineClasses/SpatialPackageMapClient.h"
+#include "LoadBalancing/AbstractLBStrategy.h"
+#include "LoadBalancing/OwnershipLockingPolicy.h"
+#include "Interop/SpatialSender.h"
+#include "Schema/SpatialDebugging.h"
+
+FSpatialLoadBalancingHandler::FSpatialLoadBalancingHandler(USpatialNetDriver* InNetDriver)
+	: NetDriver(InNetDriver)
+{
+
+}
+
+FSpatialLoadBalancingHandler::ProcessActorResult FSpatialLoadBalancingHandler::ProcessSingleActor(AActor* Actor, AActor*& OutNetOwner, VirtualWorkerId& OutWorkerId)
+{
+	const Worker_EntityId EntityId = NetDriver->PackageMap->GetEntityIdFromObject(Actor);
+	if (EntityId == SpatialConstants::INVALID_ENTITY_ID)
+	{
+		return None;
+	}
+
+	if (!Actor->HasAuthority())
+	{
+		return None;
+	}
+
+	UpdateActorSpatialDebugging(Actor, EntityId);
```

Comment:
Can we move this to `ProcessMigrations`?

Reply:
Why ? We would update the debug information for all replicated actors, not just the ones migrating (it's updating the lock info).